io-native/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/RecordChannelUtils.java

```diff
@@ -107,6 +112,23 @@ public static Object getNext(BObject channel) {
         }
     }
 
+    public static Object readRecord(BObject channel, BString separator) {
+        BufferedReader bufferedReader = (BufferedReader)
+                channel.getNativeData(BUFFERED_READER_ENTRY);
+        try {
+            String line = bufferedReader.readLine();
+            if (line == null) {
+                bufferedReader.close();
+                return IOUtils.createEoFError();
+            }
+            String[] record = line.strip().split(separator.getValue());
+            return ValueCreator.createArrayValue(StringUtils.fromStringArray(record));
+        } catch (IOException e) {
+            log.error(e.toString());
+            return IOUtils.createError(e.toString());
```

Comment:
create error should include message as well as stacktrace for easier debuggingg

Reply:
Fixed in here (https://github.com/ballerina-platform/module-ballerina-io/pull/31/commits/98acb67cdbfd15b787de2bc02c14a106f0615bfe)